Sources/VLCMediaDataSource.swift

```diff
@@ -89,6 +89,7 @@ struct VLCMediaType {
             preconditionFailure("TODO")
             return videoPlaylist
         case .allVideos:
+            getAllVideos()
```

Comment:
This fixes the test but at a very high expense. What you do here is that you call the generate all videos method for every single cell actually, because this method is called from object(at index: Int, subcategory: VLCMediaSubcategory). getAllVideos() is a very expensive method because it does fetchrequests to read all the videos out of the database with fetchrequests. 

How we would want to fix it, is by only regenerating the allVideos array when we detect that files were added or deleted. That's why there is a mediaFileDiscoverer and in the videostab we should have a method that is called with the changes that can then animate insertion or deletion of cells. That's at least how it should work ideally at the end. 
For a beginning it's enough if it refreshes when we detect videos have been added or deleted